awscli/customizations/cloudformation/package.py

```diff
@@ -112,19 +115,70 @@ class PackageCommand(BasicCommand):
         }
     ]
 
+    def _get_bucket_region(self, s3_bucket, s3_client):
+        s3_loc = s3_client.get_bucket_location(
+            Bucket=s3_bucket).get("LocationConstraint", "us-east-1")
+        return s3_loc.get
+
     def _run_main(self, parsed_args, parsed_globals):
+        region = parsed_globals.region if parsed_globals.region else "us-east-1"
         s3_client = self._session.create_client(
             "s3",
             config=Config(signature_version='s3v4'),
-            region_name=parsed_globals.region,
+            region_name=region,
             verify=parsed_globals.verify_ssl)
 
         template_path = parsed_args.template_file
         if not os.path.isfile(template_path):
             raise exceptions.InvalidTemplatePathError(
-                    template_path=template_path)
+                template_path=template_path)
+
+        if (parsed_args.s3_bucket is not None):
+            bucket = parsed_args.s3_bucket
+            s3_bucket_region = self._get_bucket_region(bucket, s3_client)
+            if not s3_bucket_region == region:
+                raise exceptions.PackageFailedRegionMismatchError(
+                    bucket_region=s3_bucket_region,
+                    deploy_region=region
+                )
+        else:
+            sts_client = self._session.create_client(
+                "sts",
+                config=Config(signature_version='s3v4'),
+                verify=parsed_globals.verify_ssl
+            )
+            account = sts_client.get_caller_identity().get('Account', "")
+            bucket = "sam-{region}-{account}".format(
+                account=account,
+                region=region
+            )
 
-        bucket = parsed_args.s3_bucket
+            # Check if SAM deployment bucket already exists otherwise create it
+            try:
+                s3_client.head_bucket(Bucket=bucket)
+            except ClientError as e:
+                if e.response["Error"]["Code"] == "404":
+                    sys.stdout.write(
+                        self.MSG_PACKAGE_S3_BUCKET_CREATION.format(
+                            bucket=bucket, region=region))
+
+                    _s3_params = {
+                        "all_regions": {
+                            "Bucket": bucket
+                        },
+                        "us_standard": {
+                            "Bucket": bucket,
+                            "CreateBucketConfiguration": {
```

Comment:
I don't understand. I thought the location constraint was for non-us-standard regions. Am I missing something?

Reply:
You're right - Fixed.